src/utils/tekton-results.ts

```diff
@@ -304,6 +304,29 @@ const getFilteredPipelineRuns = (
     cacheKey,
   );
 
+  /*
+  When pipelineruns are running, the etcd would keep their results.
+  While the tekton record would keep the conditions as:
+  "conditions": [
+    {
+      "type": "Unknown",
+      "reason": "Succeeded",
+      "status": "Running",
+    ...
+  Deleting pipelines from etcd makes the tekton record would be never updated as others.
+  So for those tekton results, it is useless to users and we need to filter them out.
+  Otherwise, these jobs would be always shown as 'Running' and bring unexpected troubles.
+  */
+  const filteredPipelineRuns = originalPipelineRuns.filter((pipelinerun) => {
+    return (
+      pipelinerun.status?.conditions?.every(
+        (c) => !(c.status === 'Unknown' && c.type === 'Succeeded' && c.reason === 'Running'),
+      ) ?? true
+    );
+  });
+  return [filteredPipelineRuns, list];
+};
```

Comment:
It's not a good idea to just filter out the `Unknown` status because When a pipeline run is running, its Condition Succeeded status is initially unknown. If the pipeline is deleted while in this state, it is archived to Tekton results with the same 'unknown' status. So we will end up filtering out the Running pipelines also.

Reply:
For the regular running pipeline, it would be shown running well.  see the screenshot:
![Screenshot 2025-02-04 at 10 57 10](https://github.com/user-attachments/assets/996c51f7-03f8-4e8d-af4c-480f4a0abf6c)
We will not filter out regular running pipelines. You could try to rerun one test run like https://localhost:8080/workspaces/wlin-tenant/applications/test-unstopped-pipeline/pipelineruns/test-28bg4 to check the whole process.

The filter will not affect any regular process.
Just when we delete the running pipeline, the pipeline would be gone  and this is the expected behavior.


BTW, do you have better idea, if you would give more suggestion, let me do more improvement with better solution. 
Is it possible we ask other team to add some item in the general status of the pipeline?

Reply:
Make sense, only Tekton results are being filtered here. can you add a few test cases for this with running status condition and other status condition.